src/Model/Group.php

```diff
@@ -0,0 +1,422 @@
+
+			$return = self::remove($gid);
+		}
+
+		return $return;
+	}
+
+	/**
+	 * @brief Adds a contact to a group
+	 *
+	 * @param int $gid
+	 * @param int $cid
+	 * @return boolean
+	 */
+	public static function addMember($gid, $cid)
+	{
+		if (!$gid || !$cid) {
+			return false;
+		}
+
+		$row_exists = dba::exists('group_member', ['gid' => $gid, 'contact-id' => $cid]);
+		if ($row_exists) {
+			// Row already existing, nothing to do
+			$return = true;
+		} else {
+			$return = dba::insert('group_member', ['gid' => $gid, 'contact-id' => $cid]);
+		}
+
+		return $return;
+	}
+
+	/**
+	 * @brief Removes a contact from a group
+	 *
+	 * @param int $gid
+	 * @param int $cid
+	 * @return boolean
+	 */
+	public static function removeMember($gid, $cid)
+	{
+		if (!$gid || !$cid) {
+			return false;
+		}
+
+		$return = dba::delete('group_member', ['gid' => $gid, 'contact-id' => $cid]);
+
+		return $return;
+	}
+
+	/**
+	 * @brief Removes a contact from a group based on its name
+	 *
+	 * @deprecated Use Group::removeMember instead
+	 *
+	 * @param int $uid
+	 * @param string $name
+	 * @param int $cid
+	 * @return boolean
+	 */
+	public static function removeMemberByName($uid, $name, $cid)
+	{
+		$gid = self::getIdByName($uid, $name);
+
+		$return = self::removeMember($gid, $cid);
+
+		return $return;
+	}
+
+	/**
+	 * @brief Returns the combined list of contact ids from a group id list
+	 *
+	 * @param array $group_ids
+	 * @param boolean $check_dead
+	 * @param boolean $use_gcontact
+	 * @return array
+	 */
+	public static function expand($group_ids, $check_dead = false, $use_gcontact = false)
+	{
+		if (!is_array($group_ids) || !count($group_ids)) {
+			return [];
+		}
+
+		$condition = '`gid` IN (' . substr(str_repeat("?, ", count($group_ids)), 0, -2) . ')';
+		if ($use_gcontact) {
+			$sql = 'SELECT `gcontact`.`id` AS `contact-id` FROM `group_member`
+					INNER JOIN `contact` ON `contact`.`id` = `group_member`.`contact-id`
+					INNER JOIN `gcontact` ON `gcontact`.`nurl` = `contact`.`nurl`
+				WHERE ' . $condition;
+			$param_arr = array_merge([$sql], $group_ids);
+			$stmt = call_user_func_array('dba::p', $param_arr);
+		} else {
+			$condition_array = array_merge([$condition], $group_ids);
+			$stmt = dba::select('group_member', ['contact-id'], $condition_array);
+		}
+
+		$return = [];
+		while($group_member = dba::fetch($stmt)) {
+			$return[] = $group_member['contact-id'];
+		}
+
+		if ($check_dead && !$use_gcontact) {
+			require_once 'include/acl_selectors.php';
+			$return = prune_deadguys($return);
+		}
+		return $return;
+	}
+
+	/**
+	 * @brief Returns a templated group selection list
+	 *
+	 * @param int $uid
+	 * @param int $gid An optional pre-selected group
+	 * @param string $label An optional label of the list
+	 * @return string
+	 */
+	public static function displayGroupSelection($uid, $gid = 0, $label = '')
+	{
+		$o = '';
+
+		$stmt = dba::select('group', [], ['deleted' => 0, 'uid' => $uid], ['order' => ['name' => 'ASC']]);
```

Comment:
```'ASC'``` will not work as parameter. The system is expecting a boolean value (```true``` for "DESC", ```false``` or no parameter at all for "ASC")

Sorry for spotting this just now. It was very early yesterday :-)